modules/azure-public-dns/a_record.tf

```diff
@@ -6,7 +6,7 @@ resource "azurerm_dns_a_record" "this" {
 
 
   resource_group_name = lower(var.resource_group_name)
-  zone_name           = var.zone_name
+  zone_name           = azurerm_dns_zone.zone.name
```

Comment:
because this dependency wasn't there terraform didn't re-create the records this should fix it going forwards